apps/server/src/modules/learnroom/uc/dashboard.uc.ts

```diff
@@ -1,22 +1,47 @@
 import { Injectable, Inject, NotFoundException } from '@nestjs/common';
-import { DashboardEntity, EntityId, GridPositionWithGroupIndex, GridPosition, SortOrder } from '@shared/domain';
-import { IDashboardRepo, CourseRepo } from '@shared/repo';
+import { AuthorisationUtils } from '@shared/domain/rules/authorisation.utils';
+import {
+	DashboardEntity,
+	EntityId,
+	GridPositionWithGroupIndex,
+	GridPosition,
+	SortOrder,
+	RoleName,
+	Course,
+} from '@shared/domain';
+import { IDashboardRepo, CourseRepo, UserRepo } from '@shared/repo';
 // import { NotFound } from '@feathersjs/errors'; // wrong import? see NotFoundException
 
 @Injectable()
 export class DashboardUc {
 	constructor(
 		@Inject('DASHBOARD_REPO') private readonly dashboardRepo: IDashboardRepo,
-		private readonly courseRepo: CourseRepo
+		private readonly courseRepo: CourseRepo,
+		private readonly authorisationUtils: AuthorisationUtils,
+		private readonly userRepo: UserRepo
 	) {}
 
-	async getUsersDashboard(userId: EntityId): Promise<DashboardEntity> {
+	async getUsersDashboard(userId: EntityId, showSubstitute: boolean): Promise<DashboardEntity> {
+		let courses: Course[];
 		const dashboard = await this.dashboardRepo.getUsersDashboard(userId);
-		const [courses] = await this.courseRepo.findAllByUserId(
-			userId,
-			{ onlyActiveCourses: true },
-			{ order: { name: SortOrder.asc } }
-		);
+		const user = await this.userRepo.findById(userId, true);
+
+		if (this.authorisationUtils.hasRole(user, RoleName.TEACHER)) {
+			[courses] = showSubstitute
+				? await this.courseRepo.findAllByUserId(userId, { onlyActiveCourses: true }, { order: { name: SortOrder.asc } })
+				: await this.courseRepo.findAllForTeacher(
+						userId,
+						{ onlyActiveCourses: true },
+						{ order: { name: SortOrder.asc } }
+				  );
+		} else {
+			[courses] = await this.courseRepo.findAllByUserId(
+				userId,
+				{ onlyActiveCourses: true },
+				{ order: { name: SortOrder.asc } }
+			);
+		}
```

Comment:
avoid code duplication

```suggestion
if (this.authorisationUtils.hasRole(user, RoleName.TEACHER) && showSubstitute) {
		  [courses] = await this.courseRepo.findAllForTeacher(
						userId,
						{ onlyActiveCourses: true },
						{ order: { name: SortOrder.asc } }
				  );
} else {
  await this.courseRepo.findAllByUserId(userId, { onlyActiveCourses: true }, { order: { name: SortOrder.asc } })
}
```

Reply:
changed